src/bp/bpfs.ts

```diff
@@ -0,0 +1,229 @@
+      } else {
+        this._printOutOfSync()
+        this._printChangeList(changeList)
+        console.log(chalk.red(`Nothing was pushed on the remote server.`))
+      }
+    } catch (err) {
+      const error = err.response ? `${err.response.statusText} (${err.response.status})` : err.message
+      this._endWithError(`Could not push changes: ${error}`)
+    }
+  }
+
+  private async _filesCount(directory: string) {
+    const files: string[] = await Promise.fromCallback(cb =>
+      glob('**/*', { cwd: directory, nodir: true, dot: true }, cb)
+    )
+    return files.length
+  }
+
+  private _processChanges(data: FileChanges) {
+    const changeList = _.flatten(data.map(x => x.changes))
+    return {
+      localFiles: _.flatten(data.map(x => x.localFiles)),
+      blockingChanges: changeList.filter(x => blockingActions.includes(x.action)),
+      changeList
+    }
+  }
+
+  private _getPushAxiosClient(archiveSize: number): AxiosInstance {
+    return axios.create({
+      baseURL: `${this.serverUrl}/api/v1/admin/versioning`,
+      headers: {
+        Authorization: `Bearer ${this.authToken}`,
+        'Content-Type': 'application/tar+gzip',
+        'Content-Disposition': `attachment; filename=archive_${Date.now()}.tgz`,
+        'Content-Length': archiveSize
+      }
+    })
+  }
+
+  private _getPullAxiosClient(): AxiosInstance {
+    return axios.create({
+      baseURL: `${this.serverUrl}/api/v1/admin/versioning`,
+      headers: {
+        Authorization: `Bearer ${this.authToken}`
+      },
+      responseType: 'arraybuffer'
+    })
+  }
+
+  private async _clearDir(destination: string): Promise<void> {
+    if (fse.existsSync(destination)) {
+      return Promise.fromCallback(cb => rimraf(destination, cb))
+    }
+  }
+
+  private _printLine({ action, path, add, del }): string {
+    if (action === 'add') {
+      return chalk.green(` + ${path}`)
+    } else if (action === 'del') {
+      return chalk.red(` - ${path}`)
+    } else if (action === 'edit') {
+      return ` o ${path} (${chalk.green('+' + add)} / -${chalk.redBright(del)})`
+    }
+    return ''
+  }
+
+  private _printOutOfSync() {
+    console.log(chalk`
+{bold Out of sync! }
+  You have changes on your file system that aren't synchronized to the remote environment.
+
+  (Replace {bold "push"} with {bold "pull"} in your command to pull remote changes on your file system)
+  (Use ${chalk.yellow('--force')} to overwrite the remote files by your local files)
+`)
+  }
+
+  private _printChangeList(changes) {
+    if (!changes.length) {
+      return
+    }
+
+    const lines = _.orderBy(changes, 'action')
+      .map(this._printLine)
+      .join('\n')
+
+    return console.log(`Differences between your local changes (green) vs remote changes (red):
+
+${lines}
+`)
+  }
+
+  private _endWithError(message: string) {
+    console.log(chalk.red(`${chalk.bold('Error:')} ${message}`))
+    process.exit()
+  }
+}
+
+export default async (argv, action) => {
```

Comment:
Not that I want you to change this because it works and would be a time loss but here are my thoughts: 

I don't see the point of creating a class for BPFS. It has a short time living and is not statefull. It would have been easy to parse the arguments in a pullChange and pushChanges functions. without the need of creating instance every time someone needs to push or pull changes.

Reply:
In this case, I find it makes the code clearer and easier to read instead of having to pass variables to each different calls. Also, it's a CLI tool, so there's no overhead of creating a new class every time. 

This is the old logic: 

![image](https://user-images.githubusercontent.com/42552874/63100607-6b93ae80-bf45-11e9-8f37-55760bec3df3.png)